server/src/main/java/org/apache/druid/server/coordinator/ReservoirSegmentSampler.java

```diff
@@ -21,36 +21,57 @@
 
 import org.apache.druid.timeline.DataSegment;
 
+import java.util.Collection;
 import java.util.List;
+import java.util.Spliterator;
 import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 final class ReservoirSegmentSampler
 {
+  final static int SPLITERATOR_SIZE_THRESHOLD = 10;
```

Comment:
I suspect this can be a lot higher without penalty. Have you done any benchmarks?

Reply:
I would say that tests are needed to check this value ensures a good uniformity of choosing a segment in a HashMap. 10 might be too little. See [this comment](https://stackoverflow.com/questions/12385284/how-to-select-a-random-key-from-a-hashmap-in-java/54893402#comment96652203_54893402) on SO.

Reply:
I ran some benchmarks by hand and observed that `25` resulted in the most uniform selection of segments. See results in this [Gist](https://gist.github.com/shivtools/7e8af2038e155e2cd8131a10f53035a5).

Reply:
did you include stats tests here? there are random components at play so you can't claim most uniform without a measure for the randomness :)

Reply:
Hey @drcrallen, I ran a very crude test that invokes `getRandomBalancerSegmentHolder()` 5000 times for different values of `spliterator size` and for each value of spliterator size, I calculated the variance (i.e `max - min`) across all segments about 200 times. I then used `.summaryStatistics()` to get an idea of what the variance looks like for each spliterator size. You can find my findings [here](https://gist.github.com/shivtools/f2766b6917dd381abd6330945a3baa49). 

In my opinion, as long as we _don't_ select `spliterator size = 2 or 3`, the variances obtained with other results look pretty similar. Is this what you had in mind?